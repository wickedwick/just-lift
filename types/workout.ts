export type Exercise = {
  id: string,
  name: string
  sets: number
  reps: number
  weight: number
  weightUnit: WeightUnit
  progressiveOverload: boolean
  overloadIncrement: number
}

export type Workout = {
  id: string
  exercises: Exercise[]
}

export type WorkoutPlan = {
  workouts: Workout[]
  daysPerWeek: number
  workoutIndex: number
  workoutInProgress: boolean
}

export type Log = {
  date: Date
  workoutId: string
  data: string
  exerciseName: string
  weight: number
  weightUnit: WeightUnit
}

export enum WeightUnit {
  Lbs,
  Kg
}
 