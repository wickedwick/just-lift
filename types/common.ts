import { Exercise, Log, WorkoutPlan } from './workout';
import { StyleProp, TextStyle, ViewStyle } from 'react-native';

export type RootStackParamList = {
  Root: undefined,
  NotFound: undefined
}

export type BottomTabParamList = {
  AutoTagger: undefined
  Celebrities: undefined
}

export type TabOneParamList = {
  BlankMenuScreen: undefined
  CreateWorkoutScreen: undefined
  SelectWorkoutScreen: undefined
  WorkoutScreen: undefined
  SettingsScreen: undefined
  ProfileScreen: undefined
  GuidedSelectionScreen: undefined
  WorkoutListScreen: undefined
}

export type TabTwoParamList = {
  TabOneScreen: undefined
  TabTwoScreen: undefined
}

export type WorkoutFormProps = {
  exercise?: Exercise
  onSubmit: (exercise: Exercise) => void
  onCancel: () => void
}

export type ExerciseCardProps = {
  exercise: Exercise
  onRemovePress: () => void
  onEditPress: () => void
}

export type ExerciseCounterProps = {
  exercise: Exercise
  setLogData: (log: Log) => void
  setWeight: (weight: number, exerciseName: string) => void
}

export type LogCardProps = {
  log: Log
}

export type ActionButtonProps = {
  onPress: () => void
  contained?: boolean
  icon?: string
  labelStyle?: StyleProp<TextStyle>
  style?: StyleProp<ViewStyle>
  text: string
}

export type RadioButtonProps = {
  label: string
  value: string
}

export type CheckboxProps = {
  label: string
  status: 'checked' | 'unchecked'
  onPress: () => void
}

export type WorkoutPlanContextProps = {
  workoutPlan: WorkoutPlan | null
  setWorkoutPlan: (workoutPlan: WorkoutPlan | null) => void
}
