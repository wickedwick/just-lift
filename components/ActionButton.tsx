import React from 'react';
import { ActionButtonProps } from '../types/common';
import { Button } from 'react-native-paper';

const ActionButton = (props: ActionButtonProps): JSX.Element => {
  return (
    <Button
      icon={props.icon}
      mode={props.contained ? 'contained' : 'text'}
      onPress={props.onPress}
      style={props.style}
      labelStyle={props.labelStyle}
    >
      {props.text}
    </Button>
  )
}

export default ActionButton
