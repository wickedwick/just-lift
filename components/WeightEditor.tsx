import ActionButton from './ActionButton'
import React, { useState } from 'react'
import { getNumberOrDefault } from '../services/utils'
import { StyleSheet, TouchableOpacity, View } from 'react-native'
import { Text, TextInput } from 'react-native-paper'
import { WeightEditorProps } from '../types/common'
import { WeightUnit } from '../types/workout'

const WeightEditor = (props: WeightEditorProps): JSX.Element => {
  const { exercise, setWeight } = props
  const [isEditingWeight, setIsEditingWeight] = useState<boolean>(false)

  return (
    <TouchableOpacity
      activeOpacity={0.6}
      onLongPress={() => { setIsEditingWeight(!isEditingWeight) }}
    >
      {isEditingWeight ? (
        <View style={styles.flexContainer}>
          <TextInput
            keyboardType={'numeric'}
            label={`Weight`}
            onChangeText={(text) => { setWeight(getNumberOrDefault(text), exercise.name) }}
            style={styles.weightTextInput}
            value={exercise.weight.toString()}
          />
          <ActionButton
            contained
            icon="check"
            labelStyle={{ fontSize: 20, textAlign: 'center' }}
            onPress={() => { setIsEditingWeight(!isEditingWeight) }}
            style={{ alignSelf: 'center', padding: 3 }}
            text=""
          />
        </View>
      ) : (
        <Text style={styles.weightLabel}>{exercise.weight} {exercise.weightUnit === WeightUnit.Kg ? 'Kg' : 'Lbs'}</Text>
      )}
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  flexContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  weightLabel: {
    fontSize: 25,
    fontWeight: 'bold',
    alignSelf: 'flex-end',
    marginBottom: 10,
    marginRight: 10,
  },
  weightTextInput: {
    marginBottom: 10,
    marginTop: 10,
    padding: 3,
    width: '75%'
  },
})

export default WeightEditor
