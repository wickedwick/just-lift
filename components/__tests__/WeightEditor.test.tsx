import ActionButton from '../ActionButton'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import WeightEditor from '../WeightEditor'
import { configure, shallow } from 'enzyme'
import { TextInput } from 'react-native-paper'
import { TouchableOpacity } from 'react-native'
import { WeightEditorProps } from '../../types/common'
import { WeightUnit } from '../../types/workout'

configure({ adapter: new Adapter() })

const props: WeightEditorProps = {
  exercise: {
    name: 'Bench Press',
    progressiveOverload: false,
    overloadIncrement: 0,
    reps: 39,
    sets: 4,
    weight: 100,
    weightUnit: WeightUnit.Lbs,
  },
  setWeight: jest.fn(),
}

describe('<WeightEditor />', () => {
  it('Renders', () => {
    const wrapper = shallow(<WeightEditor {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Handles long press', () => {
    const wrapper = shallow(<WeightEditor {...props} />)
    
    wrapper.find(TouchableOpacity).simulate('longPress')

    wrapper.setProps({ isEditingWeight: true })
    wrapper.dive().find(TextInput).simulate('changeText', '68')
    expect(props.setWeight).toHaveBeenCalled()
    
    wrapper.find(ActionButton).simulate('press')
  })
})
