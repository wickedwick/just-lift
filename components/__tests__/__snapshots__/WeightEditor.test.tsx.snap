// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeightEditor /> Renders 1`] = `ShallowWrapper {}`;
